src/components/tx-flow/flows/NewSpendingLimit/CreateSpendingLimit.tsx

```diff
@@ -54,102 +55,75 @@ export const CreateSpendingLimit = ({
-          <FormControl fullWidth sx={{ mb: 2 }}>
+          <FormControl fullWidth sx={{ mb: 3 }}>
             <AddressBookInput name="beneficiary" label="Beneficiary" />
           </FormControl>
 
-          <FormControl fullWidth sx={{ mb: 2 }}>
-            <InputLabel id="asset-label" required>
-              Select an asset
-            </InputLabel>
-            <Select
-              labelId="asset-label"
-              label={errors.tokenAddress?.message || 'Select an asset'}
-              error={!!errors.tokenAddress}
-              {...register('tokenAddress', {
-                required: true,
-                onChange: () => setValue('amount', ''),
-              })}
-              // TODO: Check when updating react-hook-form as `register` does not seem to return the value here
-              value={tokenAddress}
-            >
-              {balances.items.map((item) => (
-                <MenuItem key={item.tokenInfo.address} value={item.tokenInfo.address}>
-                  <AutocompleteItem {...item} />
-                </MenuItem>
-              ))}
-            </Select>
-          </FormControl>
+          <TokenAmountInput
+            balances={balances.items}
+            selectedToken={selectedToken}
+            maxAmount={totalAmount}
+            onMaxAmountClick={onMaxAmountClick}
+          />
 
-          <FormControl fullWidth>
-            <NumberField
-              label={errors.amount?.message || 'Amount'}
-              error={!!errors.amount}
-              required
-              {...register('amount', {
-                required: true,
-                validate: (val) => {
-                  const decimals = selectedToken?.tokenInfo.decimals
-                  return (
-                    validateAmount(val) ||
-                    validateDecimalLength(val, decimals) ||
-                    _validateSpendingLimit(val, selectedToken?.tokenInfo.decimals)
-                  )
-                },
-              })}
-            />
-          </FormControl>
-          <Typography mt={2}>
+          <Typography variant="h4" fontWeight={700} mt={3}>
+            Reset Timer
+          </Typography>
+          <Typography mb={3}>
             Set a reset time so the allowance automatically refills after the defined time period.
           </Typography>
           <FormControl fullWidth>
             <FormGroup>
-              <FormControlLabel
-                control={<Switch checked={showResetTime} onChange={toggleResetTime} />}
-                label={`Reset time (${showResetTime ? 'choose reset time period' : 'one time'})`}
-              />
+              <RadioGroup row value={!showResetTime} onChange={toggleResetTime} className={css.group}>
+                <FormControlLabel value="true" label="One Time" control={<Radio />} className={css.radio} />
+                <FormControlLabel value="false" label="Reset Time Period" control={<Radio />} className={css.radio} />
+              </RadioGroup>
```

Comment:
Not sure what we want to accomplish since this toggle already accesses the `resetTime` value that is registered in RHF. Do you have something specific in mind?

Reply:
> Not sure what we want to accomplish since this toggle already accesses the resetTime value that is registered in RHF.

These radio buttons are not registered with RHF. It is a controlled component with it's own state.

> Do you have something specific in mind?

Either `register` this of nest it inside a `Controller`.

Reply:
As per our discussion I removed the switch completely and made "One time" part of the Dropdown